### ISSUE TITLE
Remove redundant .detach

### ISSUE THREAD

**Issue by whr94621** [whr94621/NJUNMT-pytorch#20]
Some usage of ```.detach()``` is not necessary.
Such as here
https://github.com/whr94621/NJUNMT-pytorch/blob/4240f56daad1c10f53dd1a1dbdda3a958e25dfc0/src/models/dl4mt.py#L35
